bangazonapi/views/product.py

```diff
@@ -453,6 +454,34 @@ def liked(self, request):
                 {"message": str(e)}, status=status.HTTP_500_INTERNAL_SERVER_ERROR
             )
 
+    @action(methods=["delete"], detail=True)
+    def remove_from_order(self, request, pk=None):
+        """
+        @api {DELETE} /products/:id/remove-from-order DELETE product from order
+        @apiName RemoveProductFromOrder
+        @apiGroup Product
+
+        @apiHeader {String} Authorization Auth token
+        @apiHeaderExample {String} Authorization
+            Token 9ba45f09651c5b0c404f37a2d2572c026c146611
+
+        @apiParam {id} id Product Id to remove from order
+        @apiSuccessExample {json} Success
+            HTTP/1.1 204 No Content
+        """
+        try:
+            product = Product.objects.get(pk=pk)
+
+            # Remove the product from the order
+            OrderProduct.objects.filter(product=product).delete()
```

Comment:
I believe the problem may be here, where all matching products are being deleted.